### PR TITLE
preserve security capabilities

### PR DESCRIPTION
**Pull request by tofay**
rpmoci currently drops linux capabilities/extended attributes from files in the rpm. tar-rs doesn't support this natively (I'm looking into changing that), so I'm creating a pax header from the existing extended attributes.

### CODE REVIEW

File: src/oci.rs [hunk 1]
```diff
@@ -142,6 +147,75 @@ fn init_dir(layout: impl AsRef<Path>) -> Result<(), anyhow::Error> {
...
+        .filter_map(Result::ok)
+    {
+        let meta = entry.metadata()?;
+        // skip sockets as tar-rs errors when trying to archive them.
+        // For comparison, umoci also errors, whereas docker skips them
+        if meta.file_type().is_socket() {
+            continue;
+        }
+
+        let rel_path = pathdiff::diff_paths(entry.path(), src_path).unwrap();
```
Comment (dsteeley) [inline src/oci.rs:171]: entry.path() should always be under src_path but if not then this unwrap will panic. 
Could you error cleanly in that case.

File: src/oci.rs [hunk 2]
```diff
@@ -142,6 +147,75 @@ fn init_dir(layout: impl AsRef<Path>) -> Result<(), anyhow::Error> {
...
+        if entry.file_type().is_dir() || entry.file_type().is_symlink() {
```
Comment (dsteeley) [inline src/oci.rs:172]: Concerned about if a symlink points outside the oci dir layout? Bit of an edge case I would hope.

**Comment by dsteeley** [comment]
LGTM, are there any xattrs that we wouldn't want to preserve into the container image? Potential security concerns or similar?

File: src/oci.rs (hunk 1 above)
Comment (tofay) [reply src/oci.rs]: I've added an expect clarifying the assumption I'm making here.

File: src/oci.rs (hunk 2 above)
Comment (tofay) [reply src/oci.rs:177]: RPM shouldn't create symlinks that point out of the install root. If it does, then we'll copy the symlink, but not the file it was pointing to (as we don't follow symlinks), and the symlink won't work when the resulting container is run.

**Comment by tofay** [comment]
> LGTM, are there any xattrs that we wouldn't want to preserve into the container image? Potential security concerns or similar?

we want to preserve the xattr set by rpm, so as not to break security features (capabilities/selinux etc)

**Review by dsteeley — APPROVED** [review]

**Pull request merged**